### PR TITLE
Edit user class

### PR DESCRIPTION
**Pull request by turtuga**
What you think this approach to normal user class not edit your rule id and not edit admin settings?

### CODE REVIEW

File: resources/views/users/edit-add.blade.php [hunk 1]
```diff
@@ -79,6 +79,7 @@
                                 <input type="file" name="avatar">
                             </div>
 
+                            @if (Auth::user()->role_id == "1")
```
Comment (fletch3555) [inline resources/views/users/edit-add.blade.php]: I don't really like using the hard-coded role ID.  Also, I believe the role_id **should** be numerical, not a string.

File: resources/views/users/edit-add.blade.php [hunk 2]
```diff
@@ -112,13 +112,3 @@
         </div>
     </div>
 @stop
-
```
Comment (fletch3555) [inline resources/views/users/edit-add.blade.php:115]: Please explain why this was removed.  It doesn't seem related to the rest of the changes.

File: src/Http/Controllers/VoyagerUserController.php [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        // Check permission
+        Voyager::canOrFail('edit_'.$dataType->name);
+
+        $relationships = $this->getRelationships($dataType);
+
+        $dataTypeContent = (strlen($dataType->model_name) != 0)
+            ? app($dataType->model_name)->with($relationships)->findOrFail($id)
+            : DB::table($dataType->name)->where('id', $id)->first(); // If Model doest exist, get data from table name
+
+        if (Auth::user()->role_id == 2 && $dataTypeContent->role_id == 1) {
```
Comment (fletch3555) [inline src/Http/Controllers/VoyagerUserController.php]: Again, not too fond of using hard-coded role IDs.

**Comment by fletch3555** [comment]
Please explain what this PR is supposed to fix.  I can't tell from the changes alone.

File: resources/views/users/edit-add.blade.php (hunk 2 above)
Comment (turtuga) [inline resources/views/users/edit-add.blade.php]: because in add or edit user we not use tinymce or bootstraptoggle

Comment (turtuga) [reply resources/views/users/edit-add.blade.php:115]: because in add or edit user we not use tinymce or bootstraptoggle

**Comment by turtuga** [comment]
If I login with "user" in "my profile" I can change my role to administrator this is not supposed occur right ??

File: resources/views/users/edit-add.blade.php (hunk 2 above)
Comment (fletch3555) [reply resources/views/users/edit-add.blade.php:115]: Fair enough.  Just wanted to make sure it was a conscious change.

**Comment by marktopper** [comment]
In this case a user would **only** be able to change the user role if they are `admin`.

I think that it would be better to make some kind of permission for this.

**Comment by turtuga** [comment]
what do you suggest? Create one role in users section "Allow edit user class" is one option?

**Comment by marktopper** [comment]
That is my idea. But hold on, will quickly take this up on the chat with the other maintainers.

**Comment by abdgad** [comment]
@turtuga if you login with `user`, what happens if you go here: `/admin/users/1/edit`? Are you able to edit the Administrator user?

**Comment by turtuga** [comment]
I have validated `user` to not edit, but is only my opinion...

**Comment by abdgad** [comment]
@turtuga but without your modifications, you can edit other users with a normal user, correct?

**Comment by turtuga** [comment]
yes, and with normal user we can edit user class and I think it's not correct, because normal user can change user class to administrator

**Comment by marktopper** [comment]
Sorry for the late response. I have been chatting with @fletch3555 about how to solve this the proper way, and we came up with a solution for this.

It could be a option to make it possible when editing the BREAD details, to define that a certain BREAD row requires additionally permissions.
For example lets say the for the `users` table, then the permission for editing the `role_id` would have the `users_edit_role_id` permission key.

@turtuga and @abdgad, what are your thoughts on this?

**Comment by turtuga** [comment]
@marktopper yeah I like it. but we have only one little problem I think.. Imagine that one user can edit user class but he not must edit user class to administrator or class above he.

**Comment by marktopper** [comment]
/ping @fletch3555 for the proper solution for this.

**Comment by fletch3555** [comment]
I'm not sure we can officially support that use-case at this time.  For that to work, the Roles would have to be stored as a hierarchy (otherwise there's no way to know what role is "higher").  @turtuga, you're welcome to make it work yourself if it's what you need, but for now, that specific use-case will have to be put on hold until we have a chance to revamp the roles/permissions system.

**Comment by turtuga** [comment]
All right :)
